artifactory/services/utils/searchutil.go

```diff
@@ -94,12 +94,7 @@ func SearchBySpecWithAql(specFile *ArtifactoryCommonParams, flags CommonConf, re
 	}
 	if filteredReader != nil {
 		defer reader.Close()
-		fetchedProps, err = fetchProps(specFile, flags, requiredArtifactProps, filteredReader)
-		if fetchedProps != nil {
-			defer filteredReader.Close()
-			return fetchedProps, err
-		}
-		return filteredReader, err
+		reader = filteredReader
```

Comment:
Since the logic that decides which reader should be closes is not trivial, let's try to explain it with comments.